pandas/core/dtypes/inference.py

```diff
@@ -36,7 +36,7 @@
 is_iterator = lib.is_iterator
 
 
-def is_number(obj) -> TypeGuard[Number | np.number]:
+def is_number(obj: object) -> TypeGuard[Number | np.number]:
```

Comment:
Curious, is there any difference between `object` and `Any`?

Reply:
I think typeshed recommends to use `object`, when literally any object is accepted (typically for `__eq__` and similar methods) and to use `Any` when it is too difficult to write a type annotation/when the code is too dynamic.